_posts/2023-07-17-elytron-oidc-client-scope.adoc

```diff
@@ -0,0 +1,153 @@
+
+The first step is to create an `oidc.json` configuration file in the `WEB-INF` directory of the `simple-webapp-oidc` project. 
+
+Note: There is a `simple-webapp-oidc` project inside the `elytron-examples` directory. However, we will not be using that one. Instead, we will be using the slightly modified version of that inside the `elytron-oidc-client-scope` project. 
+
+There is already an `oidc.json` file inside the `WEB-INF` directory. So, we will only be editing that file by adding scope values to the existing deployment configuration. The following is the contents of the `oidc.json` file used in this example:
+```
+{
+    "client-id" : "myclient",
+    "provider-url" : "${env.OIDC_PROVIDER_URL:http://localhost:8080}/realms/myrealm",
+    "public-client" : "true",
+    "scope" : "profile email roles web-origins microprofile-jwt offline_access",
+    "principal-attribute" : "preferred_username",
+    "ssl-required" : "EXTERNAL"
+}
+```
+
+Note that we have specified the scope values to be `profile`, `email`, `web-origins`, `microprofile-jwt`, `roles` and `offline_access` in a space delimited list inside the `oidc.json` file. `profile`, `email` and `offline_access` are OpenID built-in scopes, while `web-origin`, `microprofile-jwt` and `roles` are Keycloak specific scope values and allow access to additional claims. Read the descriptions under the `client scope` tab for myclient to learn more about the functions of these scope values. 
+
+The next step is to set the auth-method to OIDC in the application’s `web.xml` file. This should have already been done, but if that is not the case, please add the following command: 
+
+```
+<login-config>
+    <auth-method>OIDC</auth-method>
+</login-config>
+```
+
+If you would like to configure the `elytron-oidc-client` subsystem instead, then follow the steps outlines in the next section. 
+
+Note: If you have both the subsystem and the deployment configured, then the `elytron-oidc-client` subsystem configuration will override the deployment configuration. 
+
+=== Subsystem Configuration
+
+To configure the server using the `elytron-oidc-client` subsystem, we will be making use of the `WILDFLY_HOME` project. 
+
+First start the server. We will be using a port offset of 10 since our Keycloak instance is already exposed on port 8080.  
+```
+    $ WILDFLY_HOME/bin/standalone.sh -Djboss.socket.binding.port-offset=10
+```
+
+Open a new terminal, navigate to the `WILDFLY_HOME` directory and enter the following command:
+```
+WILDFLY_HOME/bin/jboss-cli.sh --connect --controller=127.0.0.1:10000
+```
+We are specifying the port here because of the port offset we used earlier. Once you are connected to the server, you can configure the subsystem using the following commands:
+```
+/subsystem=elytron-oidc-client/secure-deployment=simple-webapp-oidc.war:add(resource=myclient, provider-url=http://localhost:8080/realms/myrealm, scope=profile email roles web-origins microprofile-jwt offline_access, realm=myrealm)
```

Comment:
We should use client-id instead of resource (resource was the older attribute name)

We shouldn't need realm, this is already part of the provider-url